aptos-move/block-executor/src/executor.rs

```diff
@@ -623,11 +606,60 @@ where
             // TODO: re-use the code cache.
             drop(versioned_cache);
         });
+        ret
+    }
 
+    fn make_final_result_in_parallel(
+        &self,
+        last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
+        num_txns: usize,
+    ) -> Result<Vec<E::Output>, E::Error> {
+        let _timer = PARALLEL_EXECUTOR_FINAL_RESULT_SECONDS.start_timer();
+        let skip_rest_index = Arc::new(Mutex::new(None));
+        let maybe_err = Arc::new(Mutex::new(None));
+        let mut final_results = if last_input_output.module_publishing_may_race() {
+            counters::MODULE_PUBLISHING_FALLBACK_COUNT.inc();
+            *maybe_err.lock().unwrap() = Some(Error::ModulePathReadWrite);
+            vec![]
+        } else {
+            self.executor_thread_pool.scope(|_| {
+                (0..num_txns)
+                    .into_par_iter()
+                    .map(
+                        |idx| match last_input_output.try_take_output(idx as TxnIndex) {
+                            Some(ExecutionStatus::Success(output)) => output,
+                            Some(ExecutionStatus::SkipRest(output)) => {
+                                let mut skip_index = skip_rest_index.lock().unwrap();
+                                if let Some(existing_index) = *skip_index {
```

Comment:
we can use Atomic type and have fetch min.

Reply:
Using a lock here is not expensive since in theory only one thread should access this lock. Atomic type might not be that clean as we have to initialize the atomic usize with -1 or `usize:max` instead of using `Option`. Let me know if you have something else in mind.

Reply:
yeah, with max, slightly cleaner maybe and just because it falls strictly within atomic semantics, but I agree this thing should be updated rarely, and even if not, wouldn't expect a performance hit.

Reply:
I'd go with an atomic (relaxed read). Not that important for this PR [will refactor]

Reply:
My preference would be keeping the lock without atomic as IMO setting the atomic with max is unclean.. can look to change if any strong opinion.

Reply:
Renaming the variable and using the vector size as the initialization value should be cleaner than a mutex IMO. OK with keeping the mutex tho.